python/lsst/daf/persistence/swiftStorage.py

```diff
@@ -0,0 +1,519 @@
+#!/usr/bin/env python
+
+#
+# LSST Data Management System
+# Copyright 2017 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+from future import standard_library
+standard_library.install_aliases()
+
+import copy
+import os
+import tempfile
+import time
+import yaml
+
+from . import Storage, StorageInterface, PosixStorage, ButlerLocation
+from lsst.log import Log
+
+
+class SwiftStorage(StorageInterface):
+    """Storage Interface implementation specific for Swift
+
+    Requires that the following environment variables exist:
+    SWIFT_USERNAME : string
+        The username to use when authorizing the connection.
+    SWIFT_PASSWORD : string
+        The password to use when authorizing the connection.
```

Comment:
Hmm, security through full exposure of passwords. It should work nicely :scream_cat: